nifi-mock/src/main/java/org/apache/nifi/util/MockProcessSession.java

```diff
@@ -793,13 +798,21 @@ private void validateState(final FlowFile flowFile) {
             throw new FlowFileHandlingException(flowFile + " is not the most recent version of this flow file within this session");
         }
 
+
         for (final List<MockFlowFile> flowFiles : transferMap.values()) {
             if (flowFiles.contains(flowFile)) {
                 throw new IllegalStateException(flowFile + " has already been transferred");
             }
         }
     }
 
+    private void validateDestinationRelation(final Relationship relationship) {
+        if (!relationships.contains(relationship)) {
+            rollback();
```

Comment:
This class is limited to testing purpose, so throwing an exception would make fail the associated test. I think this is what we want and in this case rollback is not needed. In a real context we should not be in such situation where the relationship does not exist.